pkg/file/seekjoiner/joiner_test.go

```diff
@@ -7,32 +7,84 @@ package seekjoiner_test
 import (
 	"bytes"
 	"context"
+	"errors"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"testing"
 
+	"github.com/ethersphere/bee/pkg/encryption/store"
+	"github.com/ethersphere/bee/pkg/file/pipeline"
+	"github.com/ethersphere/bee/pkg/file/seekjoiner"
 	joiner "github.com/ethersphere/bee/pkg/file/seekjoiner"
 	filetest "github.com/ethersphere/bee/pkg/file/testing"
 	"github.com/ethersphere/bee/pkg/storage"
 	"github.com/ethersphere/bee/pkg/storage/mock"
 	"github.com/ethersphere/bee/pkg/swarm"
+	"gitlab.com/nolash/go-mockbytes"
 )
 
+func TestJoiner_ErrReferenceLength(t *testing.T) {
+	store := mock.NewStorer()
+	joiner := joiner.NewSimpleJoiner(store)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var err error
+	_, _, err = joiner.Join(ctx, swarm.ZeroAddress)
+	if !errors.Is(err, storage.ErrReferenceLength) {
+		t.Fatalf("expected ErrReferenceLength %x but got %v", swarm.ZeroAddress, err)
+	}
+}
+
 // TestJoiner verifies that a newly created joiner returns the data stored
 // in the store when the reference is one single chunk.
 func TestJoinerSingleChunk(t *testing.T) {
```

Comment:
why not have this test with encryption too?

Reply:
this is covered in the encryption test